versioned/persist/mongodb/src/main/java/org/projectnessie/versioned/persist/mongodb/MongoDatabaseAdapter.java

```diff
@@ -632,6 +633,30 @@ protected void doWriteMultipleCommits(
     insert(client.getCommitLog(), docs);
   }
 
+  @Override
+  protected void doUpdateMultipleCommits(
+      NonTransactionalOperationContext ctx, List<CommitLogEntry> entries)
+      throws ReferenceNotFoundException {
+    List<WriteModel<Document>> requests =
+        entries.stream()
+            .map(e -> toDoc(e.getHash(), toProto(e).toByteArray()))
+            .map(
+                d ->
+                    new UpdateOneModel<Document>(
+                        Filters.eq(d.get(ID_PROPERTY_NAME)),
+                        new Document("$set", d),
+                        new UpdateOptions().upsert(false)))
+            .collect(Collectors.toList());
+
+    BulkWriteResult result = client.getCommitLog().bulkWrite(requests);
+
+    verifyAcknowledged(result, client.getCommitLog());
+
+    if (result.getMatchedCount() != entries.size()) {
+      throw new ReferenceNotFoundException("");
```

Comment:
Not sure this is the right exception type for this kind of failure. I think we ought to introduce a more specific exception for missing commits. I suppose we should expose this as an "Internal Server Error" as opposed to `NessieReferenceNotFoundException`, which would currently be the case, AFAIK.

Reply:
Why not? It means that commits to be updated do not exist.

Reply:
`Deteched` is also a reference.

Reply:
Yes, but my point is: will this error be based on a user-provided reference?

Reply:
We do not even indicate what is not found :thinking: 

Could we have a message that would allow interpreting the error in the right way if it ever shows up in API output?

Reply:
No human interaction for this call

Reply:
It still does not feel right to me to throw a `ReferenceNotFoundException` here. 

WDYT about a different exception type? This failure may occur when some commits are updated and some not, which is still different from the "normal" cases of "reference not found".

All in all, I do not feel very strongly about this. I'd be ok with current code if there no simple alternative.